datadog/dogshell/monitor.py

```diff
@@ -173,6 +207,100 @@ def _show_all(cls, args):
                                  (d["query"]),
                                  (d["type"])]))
 
+    @classmethod
+    def _pull(cls, args):
+        api._timeout = args.timeout
+        res = api.Monitor.get(args.monitor_id)
+        report_warnings(res)
+        report_errors(res)
+
+        cls._write_monitor_to_file(
+            res, args.filename,
+            args.format, args.string_ids)
+
+    @classmethod
+    def _pull_all(cls, args):
+        api._timeout = args.timeout
+
+        def _name_to_filename(name):
+            # Get a lowercased version with most punctuation stripped out...
+            no_punct = ''.join([c for c in name.lower() if c.isalnum() or c in [" ", "_", "-"]])
+            # Now replace all -'s, _'s and spaces with "_", and strip trailing _
+            return no_punct.replace(" ", "_").replace("-", "_").strip("_")
+
+        format = args.format
+        res = api.Monitor.get_all(
+            group_states=args.group_states, name=args.name,
+            tags=args.tags, monitor_tags=args.monitor_tags
+        )
+        report_warnings(res)
+        report_errors(res)
+
+        if not os.path.exists(args.pull_dir):
+            os.mkdir(args.pull_dir, 0o755)
+
+        used_filenames = set()
+        for monitor_summary in res:
+            filename = _name_to_filename(monitor_summary['name'])
+            if filename in used_filenames:
+                filename = "{0}-{1}".format(filename, monitor_summary['id'])
+            used_filenames.add(filename)
+
+            cls._write_monitor_to_file(
+                monitor_summary, os.path.join(args.pull_dir, filename + ".json"),
+                format, args.string_ids)
+        if format == 'pretty':
+            print(("\n### Total: {0} monitors to {1} ###"
+                  .format(len(used_filenames), os.path.realpath(args.pull_dir))))
+
+    @classmethod
+    def _push(cls, args):
+        api._timeout = args.timeout
+        for f in args.file:
+            try:
+                monitor_obj = json.load(f)
+            except Exception as err:
+                raise Exception("Could not parse {0}: {1}".format(f.name, err))
+
+            if 'id' in monitor_obj:
+                # Always convert to int, in case it was originally a string.
+                monitor_obj["id"] = int(monitor_obj["id"])
+                res = api.Monitor.update(**monitor_obj)
```

Comment:
Not a fan of this as it’s not very smart. For example you can’t change a monitor type but this will attempt and fail and leave the user confused.